### DISCUSSION TITLE
Issue with Interactions and Modals

### DISCUSSION THREAD

**Discussion by baremixas** [Rapptz/discord.py#9744] [Q&A] [answered]
I have my bot setup on latest 3.2.3 discord.py module, however I keep getting error while trying to use Modals.

Simple example from repo's example:

After running the following command from discord chat:

@bot.command(description="Submit feedback")
async def feedback(interaction: discord.Interaction):
&emsp; await interaction.response.send_modal(Feedback())

I get error:

discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'Context' object has no attribute 'response'

Which means, that interaction isn't of type discord.Interaction like defined in input arguments, but of type Context.
I tried to treat it as a Context object, however interaction.interaction returns None value.

**Comment by tailoric** [comment, marked as answer]
This is because what you defined here is not a slash command but a [prefix command ](https://discordpy.readthedocs.io/en/latest/ext/commands/commands.html)
Prefix commands get a context object not an interaction 
to define a slash command you need to use the [`@bot.tree.command()`](https://discordpy.readthedocs.io/en/latest/interactions/api.html#discord.app_commands.CommandTree.command) decorator

Look at the [examples of the repository](https://github.com/Rapptz/discord.py/blob/master/examples/app_commands/basic.py) for how this is supposed to work or refer to these blog post 
1. https://about.abstractumbra.dev/discord.py/2023/01/30/app-command-basics.html
2. https://about.abstractumbra.dev/discord.py/2023/01/30/app-command-examples.html

**Reply by baremixas** [reply]
Thank you!
